cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/TableSource.java

```diff
@@ -69,11 +95,10 @@ public void prepareJob(BatchSourceContext context) {
   @Override
   public void initialize(ETLStage stageConfig) throws Exception {
     super.initialize(stageConfig);
-    String schemaStr = stageConfig.getProperties().get(Table.PROPERTY_SCHEMA);
-    Preconditions.checkArgument(schemaStr != null && !schemaStr.isEmpty(), "Schema must be specified.");
-    Schema schema = Schema.parseJson(schemaStr);
-    String rowFieldName = stageConfig.getProperties().get(Table.PROPERTY_SCHEMA_ROW_FIELD);
-    rowRecordTransformer = new RowRecordTransformer(schema, rowFieldName);
+    Preconditions.checkArgument(tableConfig.schemaStr != null && !tableConfig.schemaStr.isEmpty(),
```

Comment:
Can we check this by overriding the configurePipeline method. That way, we will fail during adapter creation time rather than during runtime.